backend/docker/backend.go

```diff
@@ -0,0 +1,330 @@
+	if err != nil {
+		return exporter.Name, errors.WithStack(err)
+	}
+
+	logger := log.GetLogger(ctx)
+	logger.Debug("Exporter container created.")
+
+	if len(container.Warnings) > 0 {
+		logger.WithFields(logrus.Fields{
+			"warnings": container.Warnings,
+		}).Warning("Docker emitted warnings during container create.")
+	}
+
+	return container.ID, nil
+}
+
+func (b DockerBackend) connectToNetwork(ctx context.Context, exporter models.Exporter, cid string) error {
+	endpointSettings := network.EndpointSettings{}
+	err := b.cli.NetworkConnect(ctx, b.promNetwork, cid, &endpointSettings)
+
+	if err != nil && strings.Contains(err.Error(), "endpoint with name") {
+		return nil
+	} else if err != nil {
+		return errors.WithStack(err)
+	}
+
+	logger := log.GetLogger(ctx)
+	logger.Debug("Exporter connected to prometheus network.")
+
+	return nil
+}
+
+func (b DockerBackend) startContainer(ctx context.Context, exporter models.Exporter, cid string) error {
+	logger := log.GetLogger(ctx)
+	logger.Debug("Starting exporter container.")
+
+	err := b.cli.ContainerStart(ctx, cid, types.ContainerStartOptions{})
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	return nil
+}
+
+/* func (b DockerBackend) FindMissingExporters(ctx context.Context) error {
+	containers, err := b.cli.ContainerList(ctx, types.ContainerListOptions{})
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	containerNames := make(map[string]string, 0)
+	for _, container := range containers {
+		for _, name := range container.Names {
+			containerNames[name] = name
+		}
+	}
+
+	// Iterate over containers to find which one should have an associated
+	// exporter running but does not
+	for _, container := range containers {
+		// Ignore exporters
+		if _, ok := container.Labels[LABEL_EXPORTED_NAME]; ok {
+			continue
+		}
+
+		exporterName := getExporterName(container.Names[0])
+		if _, ok := containerNames[exporterName]; ok {
+			continue
+		}
+
+		logger := log.GetLogger(ctx).WithFields(logrus.Fields{
+			"exported.id":   container.ID,
+			"exported.name": container.Names[0],
+		})
+		ctx := log.WithLogger(ctx, logger)
+
+		err := b.handleContainerStart(ctx, container.ID, b.promNetwork)
+		if err != nil {
+			logger.Errorf("%+v", err)
+		}
+	}
+
+	return nil
+} */
+
+func (b DockerBackend) CleanupExporters(ctx context.Context, force bool) error {
+	exporters, err := b.cli.ContainerList(ctx, types.ContainerListOptions{
+		All: true,
+		Filters: filters.NewArgs(
+			filters.Arg("label", LABEL_EXPORTED_ID),
+		),
+	})
+
+	if err != nil {
+		return errors.WithStack(err)
+	}
+
+	logger := log.GetLogger(ctx)
+	logger.Debugf("Found %d running exporters.", len(exporters))
+
+	failed := []string{}
+	for _, exporter := range exporters {
+		err := b.stopExporter(ctx, exporter, force)
+		if err != nil && !IsErrExportedTaskStillRunning(err) {
+			return err
+		}
+		if err != nil && IsErrExportedTaskStillRunning(err) {
+			failed = append(failed, exporter.Names[0])
+		}
+	}
+
```

Comment:
A message specifying how much exporters cleanup failed/succeeded should be logged.